### PR TITLE
Refining user bulk load documentation

### PR DESCRIPTION
**Pull request by esthermmoturi**
From [this](https://forum.communityhealthtoolkit.org/t/bulk-user-upload-chws-and-supervisors/2161/19) forum post, it is evident that there is a huge gap between documentation on Bulk user upload and what actually works.The goal of this PR is to make the documentation:

- More detailed(spelt out step by step guide) to ensure users find it easy to upload users in bulk
- Less technical.Since the users of the feature vary from technical to non technical people, the documentation should be less technical
- Less scattered. To ensure one is able to follow the documentation step by step, explanation on what each part of the template spreadsheet does, should be clustered to avoid the user missing out on what the significance of each section is.
- Improve wording to ensure ease in reading

### CODE REVIEW

**Comment by garethbowen** [comment]
@esthermmoturi Is this ready for review?

**Comment by esthermmoturi** [comment]
@garethbowen I am working to get it ready by end of this week.

**Comment by esthermmoturi** [comment]
@mrjones-plip , any luck reviewing this pull request?

**Comment by mrjones-plip** [comment]
Hey @esthermmoturi !  Sorry for the delay in getting back to. Also - thanks for taking some time to improve our bulk user upload docs!

I'm going to use this PR to improve your process of updating the CHT docs.  I suspect it will ttake longer, but hopefully be helpful in the long run.  

Every PR should have a goal. Often that's in the form of a ticket. For example, see "[fully offline CHT deployment](https://github.com/medic/cht-docs/issues/441)". If it's just a PR (that's fine!), we need to put the goals in the PR.  Goals helps for two reasons:
* The reviewer knows what to look for, what  to compliment on, what to give constructive criticism on
* The author can review their work to make sure it's achieving the goals they stated


My request is for you to follow these guidelines and review your own changes in this PR.:

1. Write a goal or three in [the body](https://github.com/medic/cht-docs/pull/1115#issue-1752422394) of the PR.  Maybe something like, "This PR adds more explicit steps on matching the user upload gsheet template to the actual deployed app config" or maybe "This PR removes confusion around ambiguous terms of 'update' vs 'upload' vs 'create'" (these are both hypothetical!"
2. Consider that if your change is just an opinion with out adding any facts or details, don't make the change.  So if the original text was "This is how to write an input form for the CHT" and you change it to "After reading this document, you'll know how to write an input form for the CHT" - just keep the original text, you're not adding anything new.
3. If you remove content, have a reason.  For example, if we have "nginx uses memcachd to store often used static binary files often referenced by HTTPS2 requests" - that's probably too detailed! We could safely say , "Our web server, nginx, has caching to improve performance when serving images over HTTP2"
4. In general, try to write as concisely as possible.   If the same thing can be said in fewer words, go for it! Have a good reason (per above) to make something longer when making a change.

To make it this more actionable, here's some examples in your PR where you can improve based on the guidelines above:

1. Write a goal:
   * "Refine" is to vauge
2. change is just an opinion:
   * I'm not sure the [sub-title change is better](https://github.com/medic/cht-docs/pull/1115/files#diff-2c65ce5c4cf3f1c29f81cc482a9bd73bb97b4b7eaaeb4f976b8295d0ef6ffb6eR6), it feels like an opinion
   * I think you're trying to [standardize on "Import" vs "Upload" ](https://github.com/medic/cht-docs/pull/1115/files#diff-2c65ce5c4cf3f1c29f81cc482a9bd73bb97b4b7eaaeb4f976b8295d0ef6ffb6eR19)- not an opinion, an improvement! Be sure you do it through out the doc
3. remove content, have a reason
    * I think it's important to [keep the explicit path](https://github.com/medic/cht-docs/pull/1115/files#diff-2c65ce5c4cf3f1c29f81cc482a9bd73bb97b4b7eaaeb4f976b8295d0ef6ffb6eL35) of the API and that it's a `POST`.  
    *  linking to "this" is always ambiguous and we should link to explicit terms so folks know what they're getting, like this [W3C post on the topic](https://www.w3.org/QA/Tips/noClickHere). this was removed in the above edit
    * I think the [page info box at the top](https://github.com/medic/cht-docs/pull/1115/files#diff-2c65ce5c4cf3f1c29f81cc482a9bd73bb97b4b7eaaeb4f976b8295d0ef6ffb6eL25) is important as it calls out facts readers shouldn't miss
4. write as concisely
   * I think [these two lines](https://github.com/medic/cht-docs/pull/1115/files#diff-2c65ce5c4cf3f1c29f81cc482a9bd73bb97b4b7eaaeb4f976b8295d0ef6ffb6eR47), could be condensed to:
   >  Before using bulk user upload , please familiarize yourself with the 3 sections of the spreadsheet:

**Comment by esthermmoturi** [comment]
@mrjones-plip ,Thank you for the recommendation ,I will work on the issues you have raised.

**Comment by esthermmoturi** [comment]
Hey @mrjones-plip , while I agree with some of the pointers indicated above, a couple of items have caught my attention:

1. I am standardizing the general framework of documentation and although the page info is a good way to summarize what is on the page, it adds no value and the outcome of the documentation is already highlighted in the first few lines of the documentation.If necessary, a summary of the feature can be placed under page description and will appear [here](https://docs.communityhealthtoolkit.org/apps/guides/data/).

> I think the [page info box at the top](https://github.com/medic/cht-docs/pull/1115/files#diff-2c65ce5c4cf3f1c29f81cc482a9bd73bb97b4b7eaaeb4f976b8295d0ef6ffb6eL25) is important as it calls out facts readers shouldn't miss

2. Our primary readers are semi and non-technical and to them wording makes a huge difference

> Consider that if your change is just an opinion with out adding any facts or details, don't make the change. So if the original text was "This is how to write an input form for the CHT" and you change it to "After reading this document, you'll know how to write an input form for the CHT" - just keep the original text, you're not adding anything new.

3. As I work on documentation changes, I follow the available documentation step by step and that is how I notice gaps. For example, for this documentation, explanations were scattered and a lot of things were assumed as evidenced from continuous contribution to [this](https://forum.communityhealthtoolkit.org/t/bulk-user-upload-chws-and-supervisors/2161) forum post.
 Unless a feature is extremely customizable or has many moving parts, documentation should guide a user from beginning to end without them having to ask for guidance.

4. I agree with adding goals on the PR. This is a fantastic suggestion
5.  I agree and will work on some of the removed content

> 
I think it's important to [keep the explicit path](https://github.com/medic/cht-docs/pull/1115/files#diff-2c65ce5c4cf3f1c29f81cc482a9bd73bb97b4b7eaaeb4f976b8295d0ef6ffb6eL35) of the API and that it's a POST.
linking to "this" is always ambiguous and we should link to explicit terms so folks know what they're getting, like this [W3C post on the topic](https://www.w3.org/QA/Tips/noClickHere). this was removed in the above edit

**Comment by mrjones-plip** [comment]
@esthermmoturi  - wow - you rock! Thanks so much for digging into my feedback in earnest.  I can see you put a lot of thought and effort into this.  I really appreciate your time.

Once you've made some your changes to the PR, I'll give it another review.

Otherwise, diving into some of your response below. 

----
>>  I think the [page info box at the top](https://github.com/medic/cht-docs/pull/1115/files#diff-2c65ce5c4cf3f1c29f81cc482a9bd73bb97b4b7eaaeb4f976b8295d0ef6ffb6eL25) is important as it calls out facts readers shouldn't miss
>
> I am standardizing the general framework of documentation and although the page info is a good way to summarize what is on the page, it adds no value and the outcome of the documentation is already highlighted in the first few lines of the documentation.If necessary, a summary of the feature can be placed under page description and will appear [here](https://docs.communityhealthtoolkit.org/apps/guides/data/).

I'm not sure I agree with this standardization to remove this page info box.  On many other pages we use this a way to call out which CHT version is needed to use the feature or what the primary use case is for the doc page. If a reader misses these, they could be trying to implement the wrong solution on a version of the CHT that doesn't support the feature:

image

Maybe we can greatly simplify the one on this page?  If we keep some of the text you moved out , but still call out the version and google doc requirements - does that work?

image

>>   Consider that if your change is just an opinion with out adding any facts or details, don't make the change. So if the original text was "This is how to write an input form for the CHT" and you change it to "After reading this document, you'll know how to write an input for
>
> Our primary readers are semi and non-technical and to them wording makes a huge difference

Gotcha!  Re-reading your changes, I can see where you're going with this.  Also noting you're fixing a number of typos as well!

> As I work on documentation changes, I follow the available documentation step by step and that is how I notice gaps. For example, for [this](https://forum.communityhealthtoolkit.org/t/bulk-user-upload-chws-and-supervisors/2161) documentation, explanations were scattered and a lot of things were assumed as evidenced from continuous contribution to this forum post.
>
> Unless a feature is extremely customizable or has many moving parts, documentation should guide a user from beginning to end without them having to ask for guidance.

Makes sense - I think redoing something step by step is one of, if not the, best way to confirm docs make sense. Thanks for your effort!

> I agree with adding goals on the PR. This is a fantastic suggestion
> 
>  I agree and will work on some of the removed content

yay!  thanks again for being receptive to feedback.

**Comment by esthermmoturi** [comment]
@mrjones-plip , I have made a couple of changes following your suggestions and from my experience importing users. Where can I get as list of what each field means? I had a hard time differentiating between `contact.type`, `contact.contact_type` , `place.type` , `place.contact_type` and `type`. Are there any redundant fields that are not mapped(For creating user accounts, their contacts and places)?

**Comment by mrjones-plip** [comment]
@esthermmoturi - tl;dr - those fields are used if they're present in the spreadsheet.  They represent the different names in the hierarchy for specific types of contacts.  

@m5r - do you have some time next week - can you do a pass at defining what each of these fields is used for?

**Comment by m5r** [comment]
@esthermmoturi I dug into the cht code base and this is what I've found concerning the 5 fields you mentioned:

| field | description |
|-------|-------------|
| `{contact,place}.type`                | when creating a user, this is always supposed to be `contact`, other possible values are `district_hospital`, `health_center`, `clinic`, and `person` |
| `{contact,place}.contact_type` | this is specific to the deployment's configuration, it should be one of the contact types defined in the config's `contact_types` field |
| `type`                                            | this field is deprecated, use `roles` instead |

I grouped up `contact` and `place` because they're built the same way, the essentially have the same properties. Let me know if you have any other questions

cc @mrjones-plip

**Comment by esthermmoturi** [comment]
Thanks @m5r , the descriptions were helpful. I have added the descriptions. @m5r and @mrjones-plip Could you kindly review this PR for anything that I might need to add/omit? Thank you for the assistance so far.

**Review by mrjones-plip — CHANGES_REQUESTED** [review]
We're getting closer!  Requesting some changes to tighten things up, fix a mistake or two.

File: content/en/apps/guides/data/users-bulk-load.md [hunk 1]
```diff
@@ -13,67 +13,79 @@ relatedContent: >
...
 
-Copy in error messages here after importing. There are three fields:
+This area contains three columns that are pasted after running an import as shown in the importing users section below.
 1. `import.status:excluded`: This field can have three values. Over time, they should all be `imported` or `skipped` as you will have processed all users on the list:
     <ol type="a">
       <li><code>imported</code> - This user has already been successfully imported</li>
       <li><code>skipped</code> - This user was skipped</li>
-      <li><code>error</code> - There was an error importing see import.message:excluded field for more information</li>
+      <li><code>error</code> - Contains errors that were encountered during importation. There was an error importing see import.message:excluded field for more information</li>
     </ol>
```
[the remark this answers is not in the document]
Comment (m5r) [reply content/en/apps/guides/data/users-bulk-load.md:59]: I don't remember what was the intention behind this list's format. In the current state it's an ordered list with letters `a.` `b.` `c.` and markdown can't do that natively, hence the use of HTML here.  
I don't feel strongly either way

**Review by m5r — APPROVED** [review]
Nothing to add to Ashley's review, fantastic work Mariachana :+1:

File: content/en/apps/guides/data/users-bulk-load.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrjones-plip) [reply content/en/apps/guides/data/users-bulk-load.md:59]: ah -  I totally missed that this is an `<ol/>` and not a `<ul/>` - good catch @m5r! Knowing that, this seems benign enough to be an  `<ul/>` and use the native markup.

[the remark this answers is not in the document]
Comment (m5r) [reply content/en/apps/guides/data/users-bulk-load.md:59]: Yes, totally, go for it!

File: content/en/apps/guides/data/users-bulk-load.md [hunk 2]
```diff
@@ -13,67 +13,79 @@ relatedContent: >
...
+We will use the second use case to create user accounts and their contacts in the example below.
+## Contact Spreadsheet Instructions
 
-### Spreadsheet
+The spreadsheet interfaces with the [`POST /api/v1/users` API]({{< relref "apps/reference/api#get-apiv1users" >}}) which works as though passing a JSON array of users. Rows in the spreadsheet represent a user while columns represent properties of the user.
+Each column in the spreadsheet maps to an object property understood by the API to insert the users into the database. These properties can be found in [the Users API documentation]({{<relref "apps/reference/api#post-apiv1users" >}}).
 
-Before using the bulk user upload, please familiarize yourself with the spreadsheet used to manage the users before importing to the CHT.
-Specifically, there are three sections to the sheet:
+Contact spreadsheets are named according to the user role, for example when creating users who are chws and others who are chw_supervisors, the following contact spreadsheets are populated respectively; "contact.chw" and "contact.chw_supervisor".
```
[the remark this answers is not in the document]
Comment (esthermmoturi) [reply content/en/apps/guides/data/users-bulk-load.md:45]: Thanks for the semi colon  use suggestion, I'll look through it.

File: content/en/apps/guides/data/users-bulk-load.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (esthermmoturi) [reply content/en/apps/guides/data/users-bulk-load.md:59]: @mrjones-plip , so should we change this?

File: content/en/apps/guides/data/users-bulk-load.md [hunk 3]
```diff
@@ -13,67 +13,79 @@ relatedContent: >
...
+3. `User-Place-Type:excluded`: this column contains the type of user place to be created. This is a dropdown that is populated from the items indicated in the place vlookup spreadsheet
+4. `contact.first_name`: this column contains the first name of the user to be created
+5. `contact.last_name`: this column contains the last name of the user to be created
+6. `contact.sex`: this column contains the gender of the user to be created
+7. `contact.phone`: this column contains the phone number of the user to be created
+8. `email`: this column contains the email of the user to be created. This is an optional field
+9. `contact.meta.created_by`: this column contains metadata on the person who created the user. This is an optional field
+10. `token_login`: this column indicates whether the user should be sent login credentials via SMS as a link
+11. `contact.type` or `place.type`: this column when creating a user, is logged as contact, other possible values are district_hospital, health_center, clinic, and person
```
[the remark this answers is not in the document]
Comment (esthermmoturi) [reply content/en/apps/guides/data/users-bulk-load.md]: @mrjones-plip , is it just me who can't see the next suggestion?

File: content/en/apps/guides/data/users-bulk-load.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrjones-plip) [reply content/en/apps/guides/data/users-bulk-load.md:59]: yups!

File: content/en/apps/guides/data/users-bulk-load.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrjones-plip) [reply content/en/apps/guides/data/users-bulk-load.md]: sorry - it's [this one](https://github.com/medic/cht-docs/pull/1115#discussion_r1454656454)

[the remark this answers is not in the document]
Comment (esthermmoturi) [reply content/en/apps/guides/data/users-bulk-load.md]: Thanks, I see it.

File: content/en/apps/guides/data/users-bulk-load.md [hunk 4]
```diff
@@ -13,67 +13,79 @@ relatedContent: >
...
+There are three sections to the contact spreadsheet:
 
 ![bulk user import spreadsheet with areas labeled](users-spreadsheet.png)
 
 #### **Spreadsheet Area 1**
 
-Copy in error messages here after importing. There are three fields:
+These three columns are where you paste the results after running an import. See step 8 in "Importing users example" [below](#importing-users-example).
 1. `import.status:excluded`: This field can have three values. Over time, they should all be `imported` or `skipped` as you will have processed all users on the list:
     <ol type="a">
```
Comment (esthermmoturi) [inline content/en/apps/guides/data/users-bulk-load.md:55]:
```suggestion
    <ul >
```

Comment (esthermmoturi) [inline content/en/apps/guides/data/users-bulk-load.md:55]:
```suggestion
    <ul>
```

File: content/en/apps/guides/data/users-bulk-load.md [hunk 5]
```diff
@@ -13,67 +13,79 @@ relatedContent: >
...
       <li><code>imported</code> - This user has already been successfully imported</li>
       <li><code>skipped</code> - This user was skipped</li>
-      <li><code>error</code> - There was an error importing see import.message:excluded field for more information</li>
+      <li><code>error</code> - Contains errors that were encountered during importation. There was an error importing see import.message:excluded field for more information</li>
     </ol>
```
Comment (esthermmoturi) [inline content/en/apps/guides/data/users-bulk-load.md:59]:
```suggestion
    </ul>
```

File: content/en/apps/guides/data/users-bulk-load.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (esthermmoturi) [reply content/en/apps/guides/data/users-bulk-load.md]: Done in latest commits

**Review by mrjones-plip — APPROVED** [review]
Thanks for working through all these improvements @esthermmoturi !   These are good to go.

One minor fix, if you want, is [to make this list](https://github.com/medic/cht-docs/pull/1115#discussion_r1454647532) markdown instead of HTML.   I wasn't suggesting we move the from `<ol>` -> `<ul>` but that they be markdown.  No biggie - change if you want - but it's good to go as is!

**Pull request merged**